pkgs/test_core/lib/src/runner.dart

```diff
@@ -160,9 +160,7 @@ class Runner {
           throw ApplicationException('No tests match $patterns.');
         }
 
-        // Explicitly check "== true" here because [Engine.run] can return `null`
-        // if the engine was closed prematurely.
-        return success == true;
+        return _engine.passed.isNotEmpty && (success ?? false);
```

Comment:
We should emit a message if failing due to no tests running so people understand why we treated it as a failure - otherwise people will file issues because they think it should have passed

Reply:
Both the compact and expanded reporter have a message already. We could print something to `stderr` here I suppose?

Reply:
Do we not have a general way of logging that works with any reporter?

Reply:
I don't think we do have a nice way to ask the reporters to report a message.

We can either:
- Move the `No test ran` output _out_ of the reporters and write it to stderr here.
- Assume that all reporters will have a clear enough "no tests" indication that the exit code isn't surprising - not sure if we'd really say this is the case for the json reporter today
- Have extra output to both stderr and through the reporter.

I'd lean towards the second option, I don't think this is likely to be confusing and if it does come up as a point of confusion we can address it then. What do you think?

Reply:
@jakemac53 - are you OK with this implementation for now?